jujugui/static/gui/src/app/assets/javascripts/jujulib/test-jujulib-environment.js

```diff
@@ -167,13 +174,14 @@ describe('jujulib environment manager', function() {
     });
   });
 
-  it('can create a new model', function(done) {
+  it('can create a new model providing a location', function(done) {
+    var location = {'region': 'us-east-1', 'cloud': 'aws'};
     var bakery = {
       sendPostRequest: function(path, data, success, failure) {
         assert.equal(path, 'http://example.com/v2/model/rose');
         assert.deepEqual(JSON.parse(data), {
           name: 'fnord',
-          controller: 'foo',
+          Location: location,
```

Comment:
While unlikley, because JavaScript objects are pass by reference, the value you passed in could be modified and then you're checking that value again here. It's best to either clone this object or just copy/paste the values just to be on the safe side.